inference/enterprise/workflows/core_steps/models/roboflow/object_detection.py

```diff
@@ -0,0 +1,280 @@
+            ]
+        else:
+            serialised_result = [result.dict(by_alias=True, exclude_none=True)]
+        return self._post_process_result(
+            image=image,
+            class_filter=class_filter,
+            serialised_result=serialised_result,
+        )
+
+    async def run_remotely(
+        self,
+        image: List[Dict[str, Any]],
+        model_id: str,
+        class_agnostic_nms: Optional[bool],
+        class_filter: Optional[List[str]],
+        confidence: Optional[float],
+        iou_threshold: Optional[float],
+        max_detections: Optional[int],
+        max_candidates: Optional[int],
+        disable_active_learning: Optional[bool],
+        active_learning_target_dataset: Optional[str],
+    ) -> Union[List[Dict[str, Any]], Tuple[List[Dict[str, Any]], FlowControl]]:
+        api_url = (
+            LOCAL_INFERENCE_API_URL
+            if WORKFLOWS_REMOTE_API_TARGET != "hosted"
+            else HOSTED_CLASSIFICATION_URL
+        )
+        client = InferenceHTTPClient(
+            api_url=api_url,
+            api_key=self._api_key,
+        )
+        if WORKFLOWS_REMOTE_API_TARGET == "hosted":
+            client.select_api_v0()
+        client_config = InferenceConfiguration(
+            disable_active_learning=disable_active_learning,
+            active_learning_target_dataset=active_learning_target_dataset,
+            class_agnostic_nms=class_agnostic_nms,
+            class_filter=class_filter,
+            confidence_threshold=confidence,
+            iou_threshold=iou_threshold,
+            max_detections=max_detections,
+            max_candidates=max_candidates,
+            max_batch_size=WORKFLOWS_REMOTE_EXECUTION_MAX_STEP_BATCH_SIZE,
+            max_concurrent_requests=WORKFLOWS_REMOTE_EXECUTION_MAX_STEP_CONCURRENT_REQUESTS,
+            source="workflow-execution",
+        )
+        client.configure(inference_configuration=client_config)
+        inference_input = [i["value"] for i in image]
+        results = await client.infer_async(
+            inference_input=inference_input,
+            model_id=model_id,
+        )
+        if not issubclass(type(results), list):
+            results = [results]
+        return self._post_process_result(
+            image=image, class_filter=class_filter, serialised_result=results
+        )
+
+    def _post_process_result(
+        self,
+        image: List[dict],
+        class_filter: Optional[List[str]],
+        serialised_result: List[dict],
+    ) -> List[dict]:
+        serialised_result = attach_prediction_type_info(
+            results=serialised_result,
+            prediction_type="object-detection",
+        )
+        serialised_result = filter_out_unwanted_classes(
+            serialised_result=serialised_result,
+            classes_to_accept=class_filter,
+        )
+        serialised_result = attach_parent_info(image=image, results=serialised_result)
```

Comment:
If I understand correctly for the parent ids, we attach the `parent_id` key/val directly ont he prediction object, can we do the same for `prediction_type` and `image_metadata`?  maybe all go into a `metadata` property on the prediction dict?

Reply:
let's leave as is now, as with switch into sv.Detections that would be solved